botocore/awsrequest.py

```diff
@@ -305,9 +305,11 @@ def create_request_object(request_dict):
 
     """
     r = request_dict
-    return AWSRequest(method=r['method'], url=r['url'],
-                      data=r['body'],
-                      headers=r['headers'])
+    request_object = AWSRequest(
+        method=r['method'], url=r['url'], data=r['body'], headers=r['headers'])
+    if 'context' in r:
```

Comment:
Is there ever a reason not to have context in the request dict?  I'd prefer to have less of these conditionals if possible.  I can't think of a reason why this should be optional.